main/src/library/File.flix

```diff
@@ -840,4 +840,100 @@ namespace File {
                 Err(getMessage(ex))
         }
 
+    ///
+    /// Returns a list naming the files and directories in the directory `f`.
+    ///
+    /// Does not recursively traverse the directory.
+    ///
+    pub def list(f: String): Result[List[String], String] & Impure =
+        try {
+            import new java.io.File(String): ##java.io.File & Impure as newFile;
+            import java.io.File.list(): Array[String] & Impure;
+
+            let javaFile = newFile(f);
+            let javaList = list(javaFile);
+
+            Ok(Array.toList(javaList))
+        } catch {
+            case ex: ##java.io.IOException =>
+                import java.lang.Throwable.getMessage(): String & Impure;
+                Err(getMessage(ex))
+        }
+
+    ///
+    /// Returns an Iterator naming the files and directories in the directory `f`.
+    ///
+    /// Does not recursively traverse the directory.
+    ///
+    pub def listWithIter(f: String): Result[Iterator[String], String] & Impure =
+        try {
+            import new java.io.File(String): ##java.io.File & Impure as newFile;
+            import java.io.File.toPath(): ##java.nio.file.Path & Impure;
+            import static java.nio.file.Files.newDirectoryStream(##java.nio.file.Path): ##java.nio.file.DirectoryStream & Impure;
+            import java.nio.file.DirectoryStream.iterator(): ##java.util.Iterator & Impure;
+
+            let javaFile = newFile(f);
+            let javaPath = toPath(javaFile);
+            let javaStream = newDirectoryStream(javaPath);
+            let javaIter = iterator(javaStream);
+            convertPathIter(javaIter, fullPath = false)
+        } catch {
+            case ex: ##java.io.IOException =>
+                import java.lang.Throwable.getMessage(): String & Impure;
+                Err(getMessage(ex))
+        }
+
+    ///
+    /// Returns an Iterator naming the files and directories in the directory `f`.
+    ///
+    /// Recursively traverses the directory.
+    ///
+    pub def walk(f: String): Result[Iterator[String], String] & Impure =
```

Comment:
What happens if `f` is not a directory? Does it just return `f` or an error?

Reply:
It just returns the name of `f`

Reply:
The first next() is always `f`. Is this annoying?

Reply:
I think just returning `f` is reasonable. But I'd add that to the documentation.